src/fileidentifier/fileidentifier.cpp

```diff
@@ -67,26 +72,24 @@ Error FileIdentifier::GetSubjects(Array<StaticString<cSubjectIDLen>>& subjects)
  * Private
  **********************************************************************************************************************/
 
-Error FileIdentifier::ReadSubjectsFromFile()
+void FileIdentifier::ReadSubjectsFromFile()
 {
     std::ifstream file(mConfig.mSubjectsPath);
     if (!file.is_open()) {
-        return AOS_ERROR_WRAP(Error(ErrorEnum::eRuntime, "file not found"));
+        LOG_WRN() << "Can't open subjects file, empty subjects will be used";
+
+        return;
     }
 
     std::string subject;
 
     while (std::getline(file, subject)) {
-        if (auto err = mSubjects.EmplaceBack(); !err.IsNone()) {
-            return err;
-        }
+        auto err = mSubjects.EmplaceBack();
+        AOS_ERROR_CHECK_AND_THROW("subjects count exceeds application limit", err);
```

Comment:
Not obvious error here means exceeding application limit. So, here should be more suitable message: "can't set subject" for example.